File: remove-pizza-bg.py
from PIL import Image, ImageChops
import os
from pathlib import Path

def remove_pizza_background(input_path, output_path):
    """Remove white/light backgrounds from pizza images"""
    img = Image.open(input_path).convert('RGBA')
    
    # Get image data
    data = img.getdata()
    
    # Create new image with alpha channel
    new_data = []
    for item in data:
        # item is (R, G, B, A)
        r, g, b, a = item if len(item) == 4 else (item[0], item[1], item[2], 255)
        
        # Check if pixel is "white" or very light (background)
        # White background: R>200 AND G>200 AND B>200
        # We want to remove light backgrounds but keep pizza
        if r > 200 and g > 200 and b > 200:
            # Make transparent
            new_data.append((r, g, b, 0))
        else:
            # Keep the pixel with full alpha
            new_data.append((r, g, b, 255))
    
    img.putdata(new_data)
    img.save(output_path, 'PNG')
    print(f"✓ Removed background from {Path(input_path).name}")

# Process all pizzas
uploads_dir = Path(__file__).parent / 'uploads'
pizzas = [
    'Breakfast.png',
    'four_cheese.png',
    'Hawaiian.png',
    'Pepperoni.png',
    'Spinach.png',
    'Veggie.png',
    'meal7.png'
]

print("Removing pizza backgrounds...\n")
for pizza in pizzas:
    input_path = uploads_dir / pizza
    if input_path.exists():
        remove_pizza_background(str(input_path), str(input_path))
    else:
        print(f"✗ File not found: {pizza}")

print("\nBackground removal complete!")

File: remove-pizza-bg.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const uploadsDir = path.join(__dirname, 'uploads');
const pizzas = [
  'Breakfast.png',
  'four_cheese.png',
  'Hawaiian.png',
  'Pepperoni.png',
  'Spinach.png',
  'Veggie.png',
  'meal7.png'
];

async function removePizzaBackground() {
  console.log('Removing pizza backgrounds...\n');
  
  for (const pizza of pizzas) {
    const filePath = path.join(uploadsDir, pizza);
    const tempPath = path.join(uploadsDir, `${pizza}.temp`);
    
    try {
      if (!fs.existsSync(filePath)) {
        console.log(`✗ File not found: ${pizza}`);
        continue;
      }
      
      console.log(`Processing ${pizza}...`);
      
      // Read original image with alpha
      const { data, info } = await sharp(filePath)
        .ensureAlpha()
        .raw()
        .toBuffer({ resolveWithObject: true });
      
      const { width, height, channels } = info;
      console.log(`  Dimensions: ${width}x${height}`);
      
      // Process each pixel - remove white backgrounds
      for (let i = 0; i < data.length; i += 4) {
        const r = data[i];
        const g = data[i + 1];
        const b = data[i + 2];
        
        // If pixel is white or very light (background), make transparent
        if (r > 200 && g > 200 && b > 200) {
          data[i + 3] = 0;  // Set alpha to 0 (transparent)
        } else {
          data[i + 3] = 255;  // Set alpha to 255 (opaque)
        }
      }
      
      // Write processed image
      await sharp(data, {
        raw: {
          width,
          height,
          channels: 4
        }
      })
        .png()
        .toFile(tempPath);
      
      // Replace original with processed
      fs.renameSync(tempPath, filePath);
      
      const stats = fs.statSync(filePath);
      const size = (stats.size / 1024).toFixed(2);
      console.log(`  ✓ Background removed (${size} KB)\n`);
      
    } catch (error) {
      console.error(`✗ Error processing ${pizza}:`, error.message);
      if (fs.existsSync(tempPath)) fs.unlinkSync(tempPath);
    }
  }
  
  console.log('Background removal complete!');
}

removePizzaBackground();
